### PR TITLE
feat:Added Filter Button to filter content by watched, watching & unwatched videos and optimised Admin Panel

### PR DESCRIPTION
**Pull request by KitsuneKode**
### Enhancement Feature 
 I have added a filter button to filter contents by watched, watched, and unwatched videos. Also fixed some other things. 

##### Global
- Filter content by watch status (watched, unwatched, or in progress)
- Folder-level filtering for better organization
- Video progress bars on content cards(made it working)
- Synced with SideBar 

##### Admin Panel
- When course content is uploaded, the courses in the admin panel reload themselves, syncing the updates from the database and showing updated courses
I also added a reload button just in case to trigger another re-render
- Added Video specific title and error handling to the UpdateVideos component.

#### Limitation
- Notion docs are not filtered. 
- why?; all the markAsCompleted is a null value
- how to fix: fix the handleMarkAsCompleted function 
- Folder level filter not implemented  on SideBar

### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I assure there is no similar/duplicate pull request regarding same issue

### Video

https://github.com/user-attachments/assets/1a192480-3086-4b15-8a4d-94ff523f4179

### CODE REVIEW

**Comment by KitsuneKode** [comment]
I am working on the notion doc and folder compatibility for this feature. Should be completed by today!

**Comment by hkirat** [comment]
@devsargam

**Comment by devsargam** [comment]
@KitsuneKode can you please add screen recordings?

**Comment by KitsuneKode** [comment]
> ~@KitsuneKode can you please add screen recordings?~ my bad

wait actually I will provide the updated screen recording after the class.

**Comment by KitsuneKode** [comment]
@hkirat @devsargam I fixed all issues and possible problems. Can you review this

**Comment by devsargam** [comment]
Looks decent

**Comment by devsargam** [comment]
Reviewing

File: src/components/admin/AddContent.tsx [hunk 1]
```diff
@@ -98,12 +101,12 @@ export const AddContent = ({
...
       },
     });
-    setLoading(false);
     const responseData = await response.json();
-
+    setLoading(false);
     if (response.status === 200) {
       // handle success if needed
       toast.success(responseData.message);
+      setTrigger((prev) => prev + 1); // why? trigger a re-render, this is a hack
```
Comment (devsargam) [inline src/components/admin/AddContent.tsx:109]: Very smart

**Comment by devsargam** [comment]
Thank you sir

**Comment by devsargam** [comment]
@hkirat dispatch some bounty if possible

File: src/components/Sidebar.tsx [hunk 2]
```diff
@@ -301,3 +307,25 @@ function Check({ content }: { content: any }) {
...
+      content.videoProgress?.duration !== 0
+    );
+  }
+  if (filter === 'unwatched') {
+    return (
+      content.videoProgress?.markAsCompleted === false &&
+      content.videoProgress?.duration === 0
+    );
+  }
+}
```
[the remark this answers is not in the document]
Comment (KitsuneKode) [reply src/components/Sidebar.tsx:331]: ahh, sorry totally forgot to remove it,
actually applied the logic here but it felt over engineered so reverted that,

File: src/components/admin/AddContent.tsx (hunk 1 above)
Comment (KitsuneKode) [reply src/components/admin/AddContent.tsx:109]: lmao, it was the easiest approach!

**Comment by KitsuneKode** [comment]
@devsargam  thank you so much ser, 
working on another one, should be pushed soon

**Comment by devsargam** [comment]
waiting 👀

**Comment by hkirat** [comment]
what bounty was asigned to this?

**Comment by KitsuneKode** [comment]
> @hkirat dispatch some bounty if possible

no bounty amount was assigned just was given this @hkirat

**Pull request merged**